src/CairoLib.sol

```diff
@@ -20,7 +24,7 @@ library CairoLib {
             abi.encodeWithSignature("call_contract(uint256,uint256,uint256[])", contractAddress, functionSelector, data);
 
         (bool success, bytes memory result) = CAIRO_PRECOMPILE_ADDRESS.call(callData);
-        require(success, "CairoLib: call_contract failed");
+        require(success, string(abi.encodePacked("CairoLib: call_contract failed. Result: ", result)));
```

Comment:
Might be expensive depending on the result type
Alternative increasing readibility imo (and maybe less cost but to benchmark): if using 0.8.26 you can use custom errors in require avoiding the `string(abi.encodePacked)`
https://soliditylang.org/blog/2024/05/21/solidity-0.8.26-release-announcement/

Reply:
> Not sure this is necessary ?

There are multiple different reasons why the call should fail:
- out of bounds read (wrong input)
- caller contract code not whitelisted
- forbidden kakarot reentrancy

these error messages fit in a felt so are less than 31 characters, it should not be too expensive

Reply:
Sorry for the confusion: what I was asking is if the string + abi encode was necessary not the require
Now with your answer, the fact that it is only an error code with less 31 characters, indeed it should not be too expensive.
Are you sure it will send back only error codes and not arbitrary data ?

Reply:
yes, in case of errors (in the Cairo term), no arbitrary is sent